### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by legokidlogan**
Fixes the following bugs:
- Speed powerup not re-applying speed if another addon resets the player's speed (e.g. charge weapons)
- Regen powerup ignoring its removal timer
- Regen powerup not being refreshed properly

Hotshot powerup reworks:
- Simplifies death explosion code
- Players with the powerup will be immune to the death explosion of hotshot'ed players
  - They are still affected by hotshot ignition from direct damage with SWEPs, however
- Sets the inflictor for hotshot-induced fire damage to a special entity
  - This allows for a custom killicon, proper kill attribution, and cfc_pvp won't render the powerup useless

Since hotshot is actually usable again, I also took an opportunity to examine its stats. They were atrociously high, easily killing almost everyone in spawn with just one starting death. As such, I've tweaked the config defaults, and these convar changes should be put into the CFC3 server:
```
cfc_powerups_hotshot_explosion_base_radius 20
cfc_powerups_hotshot_explosion_max_radius 750
cfc_powerups_hotshot_explosion_base_damage 5
```

### CODE REVIEW

**Comment by legokidlogan** [comment]
https://github.com/user-attachments/assets/a7053365-760f-43d3-9499-057f2f4cc574

image

The killicon is a side-on silhouette of a powerup, with a simple fire icon to the right.

File: moon/powerups/server/hotshot.moon [hunk 1]
```diff
@@ -25,6 +25,9 @@ explodeWatcher = (ply) ->
     return unless IsValid ply
     return unless ply.affectedByHotshot
 
+    powerup = ply.latestHotshotPowerup
+    return unless powerup and not powerup.expired
```
Comment (sarahsturgeon) [inline moon/powerups/server/hotshot.moon]:
```suggestion
    return unless powerup
    return if powerup.expired
```

File: moon/powerups/server/hotshot.moon [hunk 2]
```diff
@@ -33,63 +36,61 @@ explodeWatcher = (ply) ->
...
-            ent.hotshotExplosionBurningDamage = nil
+    util.BlastDamage powerup.damageInflictor, powerup.owner, ply\WorldSpaceCenter!, scaledRadius, scaledDamage
 
 hook.Add "PostPlayerDeath", "CFC_Powerups_Hotshot_OnPlayerDeath", explodeWatcher
 
 fireDamageWatcher = (ent, damageInfo) ->
     return unless IsValid ent
 
+    powerup = ent.latestHotshotPowerup
+    return unless powerup and not powerup.expired
```
Comment (sarahsturgeon) [inline moon/powerups/server/hotshot.moon]:
```suggestion
    return unless powerup
    return if powerup.expired
```

**Review by sarahsturgeon — APPROVED** [review]
How are players ignited from hotshot now? And what's the value of making a new entity for hotshot inflictor? Just killicon?

**Comment by legokidlogan** [comment]
Players get ignited when a hotshot player attacks them with a bullet-based SWEP, or if they get hit by the death explosion from hotshot-ignited player. This is the same as before, but the code is a bit cleaner since ignition is handled in one place, rather than the death explosion handling ignition separately. The other significant change is that hotshot players are now immune to death explosions, so they don't get hurt or ignited by those.

The hotshot inflictor is for the killicon, better identify hotshot damage from other sources, and make cfc_pvp not block the damage since it isn't an entityflame inflictor anymore.

**Pull request merged**